tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/domain/MetaSyntax.java

```diff
@@ -0,0 +1,52 @@
+package fi.helsinki.cs.tmc.langs.domain;
+
+public class MetaSyntax {
+    
+    private static final String BEGIN_SOLUTION = "BEGIN[ \\t]+SOLUTION";
+    private static final String END_SOLUTION = "END[ \\t]+SOLUTION";
+    private static final String SOLUTION_FILE = "SOLUTION[ \\t]+FILE";
+    private static final String SPACES = "[ \\t]*";
```

Comment:
I seem to have missed the difference between `[ \\t]*` and the `[ \\t]+`s in the other static final Strings. My suggestion was to essentially replace `"BEGIN[ \\t]+SOLUTION"` with `"BEGIN" + SPACES + "SOLUTION"` but that obviously won't work as long as one is matching with `+` and the other with `*`. You can ignore that comment :)